src/client/Microsoft.Identity.Client/Http/Headers/WwwAuthenticateParameters.cs

```diff
@@ -0,0 +1,464 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.Internal;
+using Microsoft.Identity.Client.PlatformsCommon.Factories;
+using Microsoft.Identity.Client.Utils;
+using Microsoft.Identity.Json.Linq;
+
+namespace Microsoft.Identity.Client
+{
+    /// <summary>
+    /// Parameters returned by the WWW-Authenticate header. This allows for dynamic
+    /// scenarios such as claim challenge, CAE, CA auth context.
+    /// See https://aka.ms/msal-net/wwwAuthenticate.
+    /// </summary>
+    public class WwwAuthenticateParameters
+    {
+        private static readonly ISet<string> s_knownAuthenticationSchemes = new HashSet<string>(
+                                                                              new[] {
+                                                                                       Constants.BearerAuthHeaderPrefix,
+                                                                                       Constants.PoPAuthHeaderPrefix
+                                                                              }, 
+                                                                              StringComparer.OrdinalIgnoreCase);
+
+        /// <summary>
+        /// Resource for which to request scopes.
+        /// This is the App ID URI of the API that returned the WWW-Authenticate header.
+        /// </summary>
+        [Obsolete("The client apps should know which App ID URI it requests scopes for.", true)]
```

Comment:
By moving it to a new location, you've made it impossible to review. Reviewers cannot see the changes brought to this file.